hpms/src/main/java/gov/cms/ab2d/hpms/processing/AttestationReportProcessor.java

```
@@ -0,0 +1,87 @@
+
+                Row currentRow = iterator.next();
+                if (currentRow == null) {
+                    continue;
+                }
+
+                String contractNumber = currentRow.getCell(0).getStringCellValue();
+                // This report is ordered by contract number and date DESC, so just move on to the next row if we've already seen a contract
+                if (contractIdsSeen.contains(contractNumber)) {
+                    continue;
+                }
+
+                contractIdsSeen.add(contractNumber);
+
+                Optional<Contract> contractOptional = contractService.getContractByContractId(contractNumber);
+
+                if (contractOptional.isPresent()) {
+                    Contract contract = contractOptional.get();
+                    String attestationStatus = currentRow.getCell(2).getStringCellValue();
+                    Attestation attestation = attestationService.getMostRecentAttestationFromContract(contract);
+                    if (attestationStatus.toUpperCase().equals(AttestationStatus.ATTESTED.getValue().toUpperCase())) {
+                        String attestetedDateTimeCell = currentRow.getCell(6).getStringCellValue();
+                        // Set to Eastern time since that's where HPMS is
+                        String offset = DateUtil.getESTOffset();
+                        OffsetDateTime offsetDateTime = OffsetDateTime.parse(attestetedDateTimeCell + " " + offset,
+                                DateTimeFormatter.ofPattern(ATTESTATION_OFFSET_DATE_TIME_PATTERN));
+                        attestation.setAttestedOn(offsetDateTime);
+                    } else {
+                        attestation.setAttestedOn(null);
+                    }
+
+                    attestationService.saveAttestation(attestation);
+                } else {
+                    log.warn("Contract ID {} was not found in the database during contract report processing", contractNumber);
```

Missing contract is an error condition. Let's throw an unchecked exception here, which will cause the transaction to rollback and the uploading admin to become aware of the problem.